backends/tofino/CMakeLists.txt

```diff
@@ -0,0 +1,269 @@
+    set (BFN_P4C_GIT_SHA "Unknown")
+  else()
+    set (BFN_P4C_GIT_SHA $ENV{BFN_P4C_GIT_SHA})
+  endif()
+endif()
+set (ENV{P4C_VERSION} "${BFN_P4C_VERSION} (SHA: ${BFN_P4C_GIT_SHA})")
+MESSAGE(STATUS "p4c-barefoot version: $ENV{P4C_VERSION}")
+
+# Generate the sha specific version file. It includes the GIT SHA.
+# Because this version changes frequently, we include it separately from the normal version files.
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/git_sha_version.h.in"
+  "${CMAKE_CURRENT_BINARY_DIR}/bf-p4c/git_sha_version.h" @ONLY)
+
+
+macro(get_schema_version schema_file schema_var)
+  execute_process(
+    COMMAND python3 -c "from ${schema_file} import get_schema_version;print(get_schema_version(), end='', flush=True)"
+    OUTPUT_VARIABLE __schema_version
+    RESULT_VARIABLE __schema_errcode
+    ERROR_VARIABLE __schema_errstr
+    WORKING_DIRECTORY ${BFN_P4C_SOURCE_DIR}/compiler_interfaces/schemas)
+  if (${__schema_errcode})
+    MESSAGE(FATAL_ERROR "Error retrieving ${schema_file} version ${__schema_errstr}")
+  endif()
+  set(${schema_var} ${__schema_version})
+endmacro(get_schema_version)
+# Now force cmake to rerun if any of the files that we depend on versions for
+# change: context and manifest for now
+# We generate a pair of dummy dependency files will be ignored
+set(SCHEMA_FILES
+  compiler_interfaces/schemas/context_schema.py
+  compiler_interfaces/schemas/manifest_schema.py
+  compiler_interfaces/schemas/phv_schema.py
+  compiler_interfaces/schemas/power_schema.py
+  compiler_interfaces/schemas/resources_schema.py
+  )
+foreach (f ${SCHEMA_FILES})
+  configure_file(${BFN_P4C_SOURCE_DIR}/${f} ${CMAKE_BINARY_DIR}/${f}.dep)
+endforeach()
+
+if (ENABLE_WERROR)
+  # fixme: we do want implicit-fallthrough to be an error, but we have to
+  # supress in in p4c and in generated code somehow
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wno-error=implicit-fallthrough")
+endif()
+
+# # Gtest needs additional includes, add them here.
+# if (ENABLE_GTESTS)
+#   target_include_directories(gtestp4c
+#     PRIVATE "${BFN_P4C_SOURCE_DIR}/p4c/test/frameworks/gtest/googlemock/include"
+#     PRIVATE "${BFN_P4C_SOURCE_DIR}/p4c/test/frameworks/gtest/googletest/include"
+#     PRIVATE "${P4C_BINARY_DIR}/extensions/bf-p4c/logging"
+#     PRIVATE "bf-p4c/logging"
+#     PRIVATE "${P4C_BINARY_DIR}/control-plane/"
+#   )
+#   # -Wdeprecated-copy warns in gtest, so out of our control
+#   # TODO: -Wno-unused-function warns in p4c, should be possible to fix it
+#   target_compile_options(gtestp4c PUBLIC -Wno-unused-function -Wno-deprecated-copy)
+# endif()
+
+add_custom_target(ctags-all
+  COMMAND cd ${BFN_P4C_SOURCE_DIR}/bf-p4c &&
+          ctags -R --langmap=C++:+.def,Flex:+.l,YACC:+.ypp
+                ${P4C_SOURCE_DIR}/frontends ${P4C_SOURCE_DIR}/ir
+                ${P4C_SOURCE_DIR}/midend/ ${P4C_SOURCE_DIR}/lib .
+  DEPENDS ctags-asm ctags
+  COMMENT "Building ctags")
+
+add_custom_target(cpplint-all
+  DEPENDS cpplint-asm cpplint
```

Comment:
Since bfasm isn't present in this code, there probably should not be dependencies on it here?